src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,149 @@
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+class TodoList extends React.PureComponent {
+  state = {
+    search: '',
+    todoStatus: 'all',
+    isShuffle: false,
+  }
+
+  searchHandler = (event) => {
+    this.setState({
+      search: event.target.value,
+    });
+  }
+
+  getTodosForShow = (todos) => {
+    if (!todos) {
+      return [];
+    }
+
+    const { search } = this.state;
+    const searchToLower = search.toLowerCase();
+
+    const todosWithSearchFilter = todos.filter(({ title }) => (
+      title && title.toLowerCase().includes(searchToLower)));
+
+    return this.todosCompletedFilter(todosWithSearchFilter);
+  }
+
+  todoStatusHandler = (event) => {
+    this.setState({
+      todoStatus: event.target.value,
+    });
+  }
+
+  todosCompletedFilter = (todos) => {
+    const { todoStatus } = this.state;
+
+    switch (todoStatus) {
+      case 'active':
+        return todos.filter(({ completed }) => !completed);
+
+      case 'inactive':
+        return todos.filter(({ completed }) => completed);
+
+      case 'all':
+      default:
+        return todos;
+    }
+  }
+
+  render() {
+    const { todos, onUserChange, selectedUserId } = this.props;
+    const { search, todoStatus, isShuffle } = this.state;
+    let todosForShow = this.getTodosForShow(todos);
+
+    if (isShuffle) {
+      todosForShow = randomSort(todosForShow);
+    }
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <input
+          type="text"
+          value={search}
+          onChange={this.searchHandler}
+          placeholder="search"
+        />
+
+        <select
+          name="todoStatus"
+          onChange={this.todoStatusHandler}
+          value={todoStatus}
+        >
+          <option value="all">All</option>
+          <option value="active">active</option>
+          <option value="inactive">completed</option>
+        </select>
+
+        <button
+          type="button"
+          onClick={() => this.setState({ isShuffle: Math.random() })}
+        >
+          randomSort
+        </button>
+
+        <div className="TodoList__list-container">
+          <ul className="TodoList__list">
+            {todosForShow.map(todo => (
+              <li
+                key={todo.id}
+                className={classNames('TodoList__item',
+                  {
+                    'TodoList__item--unchecked': !todo.completed,
+                    'TodoList__item--checked': todo.completed,
+                  })}
```

Comment:
1) Строка с закрывающей скобкой должна иметь такой же отступ как и строка с открывающей
2) Если один параметр функции пишется на новой строке - то каждый должен быть на отдельной строке

Поэтому либо:
```suggestion
                className={classNames(
                  'TodoList__item',
                  {
                    'TodoList__item--unchecked': !todo.completed,
                    'TodoList__item--checked': todo.completed,
                  },
                )}
```
либо
```suggestion
                className={classNames('TodoList__item', {
                  'TodoList__item--unchecked': !todo.completed,
                  'TodoList__item--checked': todo.completed,
                })}
```